### PR TITLE
PAINTROID-63: hex color input

### PR DESCRIPTION
**Pull request by SebastianGrief**
## Link to ticket
https://jira.catrob.at/browse/PAINTROID-63 

## Description
- TextColor of EditText will turn red if the input is not a valid hex value
- Added a Constant greater than the int value a color can be as a return for not a HEX value

### CODE REVIEW

File: Paintroid/src/main/res/layout/color_chooser_layout_rgbview.xml [hunk 1]
```diff
@@ -184,7 +184,34 @@
...
+            android:paddingTop="12dp">
+
+            <TextView
+                android:id="@+id/color_chooser_color_rgb_textview_hex"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:layout_gravity="center_vertical"
+                android:gravity="center_vertical"
+                android:text="@string/color_hex"
+                android:textColor="@color/pocketpaint_color_chooser_black"
```
Comment (ThomasSchwengler) [inline Paintroid/src/main/res/layout/color_chooser_layout_rgbview.xml]: This is a color for the preset buttons

File: Paintroid/src/main/res/layout/color_chooser_layout_rgbview.xml [hunk 2]
```diff
@@ -184,7 +184,34 @@
...
 
+            <EditText
+                android:id="@+id/color_chooser_color_rgb_hex"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:layout_gravity="center_vertical"
+                android:hint="#"
+                android:inputType="textNoSuggestions"
+                tools:ignore="HardcodedText"
+                android:focusable="false"/>
```
Comment (ThomasSchwengler) [inline Paintroid/src/main/res/layout/color_chooser_layout_rgbview.xml]: Are you sure about this?
This does not allow me to input any value: https://streamable.com/p4bxv

Comment (SebastianGrief) [reply Paintroid/src/main/res/layout/color_chooser_layout_rgbview.xml]: Correct. That didn't go as planned.
I'm looking for a different way to stop the autoscroll to the EditText.

File: Paintroid/src/main/res/layout/color_chooser_layout_rgbview.xml [hunk 3]
```diff
@@ -32,13 +31,14 @@
     <TableLayout
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        android:stretchColumns="1" >
+        android:stretchColumns="1"
+        android:focusable="false">
```
Comment (ThomasSchwengler) [inline Paintroid/src/main/res/layout/color_chooser_layout_rgbview.xml]: What's the reason for setting this to `android:focusable="false"`?

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/colorpicker/RgbSelectorView.java [hunk 4]
```diff
@@ -98,6 +105,55 @@ public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
...
+			}
+		});
+	}
+
+	private boolean parseInputToCheckIfHEX(String newText) {
+		if (newText.length() == 0) {
+			return false;
+		}
+
+		if (newText.length() != 9) {
```
Comment (ThomasSchwengler) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/colorpicker/RgbSelectorView.java]: No need to check if `newText.length() == 0` then ...

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/colorpicker/RgbSelectorView.java [hunk 5]
```diff
@@ -98,6 +105,55 @@ public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
...
+			@Override
+			public void beforeTextChanged(CharSequence s, int start, int count, int after) { }
+
+			@Override
+			public void onTextChanged(CharSequence s, int start, int before, int count) { }
+
+			@Override
+			public void afterTextChanged(Editable newText) {
+				if (editTextHex.getTag() == null) {
+					if (parseInputToCheckIfHEX(newText.toString())) {
```
Comment (ThomasSchwengler) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/colorpicker/RgbSelectorView.java]: This also calls `Color.parseColor`, same as the next line

**Pull request merged**